Description: Prevent double-free error while transfering songs to player.
Author: Darran Kartaschew <chewy509@lycos.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=632454
---
 src/mtp.c |    2 --
 1 file changed, 2 deletions(-)

--- gmtp.orig/src/mtp.c
+++ gmtp/src/mtp.c
@@ -1041,7 +1041,6 @@ void albumAddTrackToAlbum(LIBMTP_album_t
         tracks[found_album->no_tracks - 1] = trackinfo->item_id; // This ID is only set once the track is on the device.
         found_album->tracks = tracks;
         ret = LIBMTP_Update_Album(DeviceMgr.device, found_album);
-        g_free(tracks);
     } else {
         // New album.
         uint32_t *trackid;
@@ -1051,7 +1050,6 @@ void albumAddTrackToAlbum(LIBMTP_album_t
         albuminfo->no_tracks = 1;
         albuminfo->storage_id = DeviceMgr.devicestorage->id;
         ret = LIBMTP_Create_New_Album(DeviceMgr.device, albuminfo);
-        g_free(trackid);
     }
     if (ret != 0) {
         displayError(_("Error creating or updating album.\n(This could be due to that your device does not support albums.)\n"));
